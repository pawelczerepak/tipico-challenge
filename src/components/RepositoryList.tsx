import React from 'react';
import styled from 'styled-components';
import {Repository as RepositoryType} from '../types';
import Repository from './Repository';
import {Loader, CenteredLoaderWrapper} from './Loader';

interface Props {
    repositories: RepositoryType[];
}

const List = styled.ul`
  list-style-type: none;
  padding: 0;
  margin: 0;
  min-height: 1570px;
  position: relative;
  @media (min-width: 768px) {
    min-height: 830px;
  }
`;

export const RepositoryList: React.FC<Props> = ({repositories}) => {
    return <List>
        {repositories.length === 0 ? <CenteredLoaderWrapper><Loader/></CenteredLoaderWrapper> : null}
        {repositories.map((r, i) => <Repository key={i} repository={r}/>)}
    </List>;
};

export default RepositoryList;
