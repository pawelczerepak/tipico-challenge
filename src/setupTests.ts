import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';
import client from './apiClient';

configure({ adapter: new Adapter() });

export const mockStore = configureMockStore([thunk, axiosMiddleware(client)]);
