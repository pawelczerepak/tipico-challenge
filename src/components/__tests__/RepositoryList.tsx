import React from 'react';
import '../../setupTests';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import RepositoryList from '../RepositoryList';

describe('RepositoryList', () => {
    const repository = {
        full_name: 'owner/repo',
        owner: {login: 'login', avatar_url: 'url', html_url: 'url'},
        description: 'desc',
        html_url: 'url',
        language: 'lang',
        stargazers_count: 34,
    };
    it('capture snapshot', () => {
        expect(toJson(shallow(<RepositoryList repositories={[repository]}/>)))
            .toMatchSnapshot();
    });
});
