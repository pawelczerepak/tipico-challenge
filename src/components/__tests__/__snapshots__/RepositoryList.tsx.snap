// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryList capture snapshot 1`] = `
<styled.ul>
  <_default
    key="0"
    repository={
      Object {
        "description": "desc",
        "full_name": "owner/repo",
        "html_url": "url",
        "language": "lang",
        "owner": Object {
          "avatar_url": "url",
          "html_url": "url",
          "login": "login",
        },
        "stargazers_count": 34,
      }
    }
  />
</styled.ul>
`;
